### DISCUSSION TITLE
Checking for dead memcache server buggy?

### DISCUSSION THREAD

**Discussion by rpluem-vf** [OpenIDC/mod_auth_openidc#1230] [General]
In 

https://github.com/OpenIDC/mod_auth_openidc/blob/28557c1b84bac245730058328430fea354a97d25/src/cache/memcache.c#L229

doesn't it need to be 

```suggestion
rc = rc && (context->cache_memcache->live_servers[i]->status != APR_MC_SERVER_DEAD); 
```

because otherwise I don't see how we iterate over all servers as the loop body in https://github.com/OpenIDC/mod_auth_openidc/blob/28557c1b84bac245730058328430fea354a97d25/src/cache/memcache.c#L228-L229

is invariant as neither `i` is used nor the array pointer of `live_servers` gets incremented.

Furthermore, what should be tested? That all servers are up? In this case the check would be correct, but I guess we would like to check that a least one server is up and hence the check would need to be:

```suggestion
	int rc = FALSE;
	int i;
	for (i = 0; !rc && i < context->cache_memcache->ntotal; i++)
		rc = rc || (context->cache_memcache->live_servers[i]->status != APR_MC_SERVER_DEAD);
	return rc;
```

or

```suggestion
	int i;
	for (i = 0;  i < context->cache_memcache->ntotal; i++) {
		if (context->cache_memcache->live_servers[i]->status != APR_MC_SERVER_DEAD)
		    return TRUE;
        }
	return FALSE;
```

**Comment by zandbelt** [comment]
I believe you're right, thanks for spotting this! The error has been there since the very first commit of that code in 2017 and is fixed now in 30d9a55.
